scripts/build/gen_isr_tables.py

```diff
@@ -131,12 +131,51 @@ def parse_args():
 #endif
 """
 
+sh_isr_function = "__attribute__((section(\".shared_isr\"))) void shared_isr_"
+sh_isr_args = "const uint32_t shared_isr_"
+sh_isr_args_section = " __attribute__((section(\".shared_isr_args\")))"
+
 def get_symbol_from_addr(syms, addr):
     for key, value in syms.items():
         if addr == value:
             return key
     return None
 
+def write_shared_isr_prototypes(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_function + str(i) + "(void *args);\n")
+
+def write_shared_isr_args_forward_declare(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_args + str(i) + "_args[" +
+                str(len(shi[i]))  +"]" + sh_isr_args_section + ";\n")
+
+def write_shared_isr(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_function + str(i) +"(void *args){\n")
+            idx = 0
+            for isr in shi[i]:
+                func = str(hex(isr[1]))
+                fp.write("\t((ISR)" + func + ")((void *)((uint32_t *)args)[" +
```

Comment:
why `(void *)(uint32_t *)`?

Reply:
This is been changed to (void *)(size_t *). The initial (size_t *) cast is used so the compiler can index the args array and the (void *) cast is to match the type expected by the ISR.